app/ports/utilities/sort_by_version.py

```diff
@@ -1,13 +1,59 @@
-from distutils.version import LooseVersion
+import re
+from functools import cmp_to_key
 
 
-def cleaned_version(version):
-    return version.replace("_", ".").replace("-", ".")
+# Tries to apply the same version sorting algorithm as used by macports-base
+# https://github.com/macports/macports-base/blob/master/src/pextlib1.0/vercomp.c
+def version_compare(version1, version2):
+    # Split on all non-alphanumeric characters and delete any empty ('') elements
+    # that might occur due to consecutive non-alphanumerics
+    version1_segments = list(filter(None, re.split('[^a-zA-Z0-9]', version1)))
+    version2_segments = list(filter(None, re.split('[^a-zA-Z0-9]', version2)))
+
+    version1_len = len(version1_segments)
+    version2_len = len(version2_segments)
+
+    # Now traverse the two lists and compare segments at each index
+    i = 0
+    while i < version1_len and i < version2_len:
+        s1 = version1_segments[i]
+        s2 = version2_segments[i]
+
+        # If version1's segment is a numeric segment and version2's segment
+        # is non-numeric, consider version1 to be a newer version
+        if s1.isnumeric() and (not s2.isnumeric()):
+            return -1
+
+        # If the segments are of different types, consider version2 be newer
+        if (s1.isnumeric() and s2.isalpha()) or (s1.isalpha() and s2.isnumeric()):
```

Comment:
@jmroot: Oh, are we really using such a strange code? Shouldn't the one that's numeric (or the one that's not) at least consistently win?

Reply:
https://github.com/macports/macports-base/blob/08d87705e9cce7b1f64c061d4335695be44099f6/src/pextlib1.0/vercomp.c#L89

I had the same thought, but I could interpret it this way only.